### DISCUSSION TITLE
Certificate Issues leading to delay in publishing new module version

### DISCUSSION THREAD

**Discussion by jpomfret** [dataplat/dbatools#9041] [General]
We recently changed certificate providers and because of that installing\updating the module on a machine where the module already existed will cause the following error, and require the `-SkipPublisherCheck`

```
Install-Package: Authenticode issuer 'CN=dbatools, O=dbatools, L=Vienna, S=Virginia, C=US' of the new module 'dbatools' with version
'2.0.5' from root certificate authority 'CN=SSL.com Root Certification Authority RSA, O=SSL Corporation, L=Houston,
S=Texas, C=US' is not matching with the authenticode issuer 'CN=dbatools, O=dbatools, L=Vienna, S=Virginia, C=US' of
the previously-installed module 'dbatools' with version '2.0.2' from root certificate authority 'CN=DigiCert Assured ID
Root CA, OU=www.digicert.com, O=DigiCert Inc, C=US'. If you still want to install or update, use -SkipPublisherCheck
parameter.
```
Issue documented here: https://github.com/dataplat/dbatools/issues/9040

This will require either a lot of communication around this error and the reason for receiving it, assuring people it is safe to download this version of the module, or reverting to use the existing digicert certificate.

**Comment by andreasjordan** [comment]
Can we continue using the digicert certificate? What are the cons in doing so? I don't remember the reason for chaning the certificat.

**Reply by jpomfret** [reply]
Believe the main reason was it's difficult to prove identity and get the cert renewed which has to happen every 3 years for digicert. But SSL.com offered a 10 year cert.

We are currently investigating whether returning to digicert is an option. 

Cc @potatoqualitee

**Reply by wsmelton** [reply]
This will be expected behavior when you change the cert issuer. PowerShell is comparing the new one with the old one and sees they are not the same, therefore throws the error.

It's by design and something new users will have to deal with for the initial upgrade to get passed the cert change.
